src/Workspaces/Core/Portable/Workspace/Workspace.cs

```diff
@@ -212,7 +212,7 @@ private protected (Solution oldSolution, Solution newSolution) SetCurrentSolutio
         /// updated by <paramref name="transformation"/> to be passed to the workspace change event.</param>
         /// <returns>True if <see cref="CurrentSolution"/> was set to the transformed solution, false if the
         /// transformation did not change the solution.</returns>
-        private protected (bool updated, Solution newSolution) SetCurrentSolution(
+        internal (bool updated, Solution newSolution) SetCurrentSolution(
```

Comment:
don't love this.

Reply:
I mainly didn't hate it because it's a non-virtual method that significantly simplifies the alternative call site form.

Reply:
The only reason I didn't hate this was because so many of the other SetCurrentSolution* overloads are internal. Which we still need to clean up, but it's not like this method was the only thing preventing a potential design problem.